### DISCUSSION TITLE
How can I change my environment if my nodes have different root passwords

### DISCUSSION THREAD

**Discussion by vasilytray** [autobase-tech/autobase#271] [Show and tell]
How can I change my environment if my nodes have different root passwords?
And when I check the connection to other nodes, I naturally get this picture

`root@ubuntu22:~/postgresql_cluster# ansible all -m ping
192.168.171.1 | SUCCESS => {
    "changed": false,
    "ping": "pong"
}
192.168.171.3 | UNREACHABLE! => {
    "changed": false,
    "msg": "Data could not be sent to remote host \"192.168.171.3\". Make sure this host can be reached over ssh: ssh: connect to host 192.168.171.3 port 22: No route to host\r\n",
    "unreachable": true
}
192.168.171.2 | UNREACHABLE! => {
    "changed": false,
    "msg": "Invalid/incorrect password: Warning: Permanently added '192.168.171.2' (ED25519) to the list of known hosts.\r\nPermission denied, please try again.",
    "unreachable": true
}
`

**Comment by vasilytray** [comment]
In addition, having solved the problem of connecting to all nodes by issuing a key (although the original question remains open), I get this picture...  PostgreSQL has not risen on any node
![Скриншот 03-03-2023 190358](https://user-images.githubusercontent.com/29880990/222716540-c9e3aab0-ab63-4b8c-82bc-38d36a76766c.jpg)

**Reply by vitabaks** [reply]
> How can I change my environment if my nodes have different root passwords?

method 1

specify the `ansible_user` and `ansible_ssh_pass` parameters at the level of each host instead of denoting the general parameters at the level of the "all" group

method 2

Use `ansible_ssh_private_key_file`

**Reply by vitabaks** [reply]
> PostgreSQL has not risen on any node

Try to update operating system on your target servers before deploying

**Reply by vasilytray** [reply]
Etcd services are run successfully. The first,  must I kill all deployed processes, before deploying again?

**Reply by vitabaks** [reply]
No. Because Ansible provides atomicity.

**Reply by vasilytray** [reply]
Thenx!

**Reply by vasilytray** [reply]
why i can`t remove claster?
root@pgnode01:~/postgresql_cluster# ansible-playbook remove_cluster.yml

PLAY [postgres_cluster] *************************************************************************************************************************************************
TASK [Gathering Facts] **************************************************************************************************************************************************ok: [192.168.171.1]
ok: [192.168.171.2]
ok: [192.168.171.3]

TASK [Stop Patroni service] *********************************************************************************************************************************************fatal: [192.168.171.1]: FAILED! => {"changed": false, "msg": "Could not find the requested service patroni: host"}
fatal: [192.168.171.2]: FAILED! => {"changed": false, "msg": "Could not find the requested service patroni: host"}
fatal: [192.168.171.3]: FAILED! => {"changed": false, "msg": "Could not find the requested service patroni: host"}

PLAY RECAP **************************************************************************************************************************************************************192.168.171.1              : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
192.168.171.2              : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
192.168.171.3              : ok=1    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   

root@pgnode01:~/postgresql_cluster# etcdctl member list
ea8843cada5e9fa: name=pgnode02 peerURLs=http://192.168.171.2:2380 clientURLs=http://192.168.171.2:2379 isLeader=true
5b709a3dc6117745: name=pgnode03 peerURLs=http://192.168.171.3:2380 clientURLs=http://192.168.171.3:2379 isLeader=false
d59e98087eb7e712: name=pgnode01 peerURLs=http://192.168.171.1:2380 clientURLs=http://192.168.171.1:2379 isLeader=false

root@pgnode01:~/postgresql_cluster# patronictl -c /etc/patroni.yml list
Command 'patronictl' not found, but can be installed with:
apt install patroni

**Reply by vitabaks** [reply]
Hi @vasilytray, thank you for informing me

fixed https://github.com/vitabaks/postgresql_cluster/commit/7924d467a85a4df4d90e25acd2782ce2407286e4

**Reply by vasilytray** [reply]
thanks

**Comment by vasilytray** [comment]
please tell me where the IP is specified for Keepalived to install a scheme of type A

**Reply by vitabaks** [reply]
`cluster_vip` variable.

https://github.com/vitabaks/postgresql_cluster/blob/master/vars/main.yml#L10

P.S.

Please create different issues/discussions for different topics and errors.
